source/fundamentals/data-formats/document-data-format-extended-json.txt

```diff
@@ -119,146 +119,280 @@ corresponds to the format of the example you want to see:
+      optionally pass a :java-docs:`JsonWriterSettings </apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+      instance to specify options such as the Extended JSON format. By default, the
+      ``JsonWriter`` uses the "Relaxed" Extended JSON format. MongoDB driver's
+      document classes also use this class to convert BSON to Extended JSON.
+      
+      The following code example shows how you can use ``JsonWriter`` to create an
+      Extended JSON string and output it to ``System.out``. We specify the format
+      by passing the :java-docs:`outputMode() </apidocs/bson/org/bson/json/JsonWriterSettings.Builder.html#outputMode(org.bson.json.JsonMode)>`
+      builder method the ``JsonMode.EXTENDED`` constant:
+      
+      .. code-block:: java
+      
+         JsonWriterSettings settings = JsonWriterSettings.builder().outputMode(JsonMode.EXTENDED).build();
+      
+         try (JsonWriter jsonWriter = new JsonWriter(new BufferedWriter(new OutputStreamWriter(System.out)), settings)) {
+             jsonWriter.writeStartDocument();
+             jsonWriter.writeObjectId("_id", new ObjectId("507f1f77bcf86cd799439012"));
+             jsonWriter.writeInt64("myNumber", 11223344);
+             jsonWriter.writeEndDocument();
+             jsonWriter.flush();
+         }
+      
+      The output of this code example resembles the following:
+      
+      .. code-block:: none
+         :copyable: false
+      
+         {"_id": {"$oid": "507f1f77bcf86cd799439012"}, "myNumber": {"$numberLong": "11223344"}}
+
+   .. tab:: JsonObject Class
+      :tabid: jsonobject-class-write
+
+      The ``JsonObject`` class is a wrapper for JSON strings, you can access the
+      string contained in a ``JsonObject`` instance using the ``toJson()`` method.
+
+      The following code snippet shows how to read an extended JSON string from a ``JsonObject``
+      using the ``toJson()`` method:
+
+      .. code-block:: java
+      
+         String ejsonStr = "{ \"_id\": { \"$oid\": \"507f1f77bcf86cd799439011\"}," +
+                           "\"myNumber\": {\"$numberLong\": \"4794261\" }}}";
+         JsonObject jsonObj = new JsonObject(ejsonStr);
+         System.out.println(jsonObj.getJson());
+
+      The output of this code example resembles the following:
+      
+      .. code-block:: none
+         :copyable: False
+      
+         { "_id": { "$oid": "507f1f77bcf86cd799439011"},"myNumber": {"$numberLong": "4794261" }}}
+
+      The ``JsonObject`` class may not seem very useful at first glance, but its usefulness
+      is easier to see when looked at in the context of the
+      :java-docs:`default codec registry <apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html?is-external=true#getDefaultCodecRegistry()>`
+      and the :java-docs:`JsonObjectCodec <apidocs/bson/org/bson/codecs/JsonObjectCodec.html>` class.
+      Below is an example based off of the :doc:`quickstart </fundamentals/data-formats/codecs>`
```

Comment:
You can tidy up the verb phrase:
```suggestion
      Below is an example based on the :doc:`quickstart </fundamentals/data-formats/codecs>`
```

Reply:
Thanks for the note, updated in both places this 'off of' language was used. I agree 'on' is cleaner and easier to understand.